### ISSUE TITLE
False positive for rule 942200 by Googlebot after upgrading to v4.24.0

### ISSUE THREAD

**Issue by mhoran** [coreruleset/coreruleset#4502]
### Description

After upgrading to CRS v4.24.0 I saw that Googlebot was being rejected by modsecurity due to rule 942200. Downgrading to v4.23.0 resolves. This could be due to #4476.

### Logs

```
---hNXNazwk---A--
[01/Mar/2026:20:41:55 -0500] 177241571537.088552 66.249.68.130 0 unix:/var/run/nginx/nginx.sock 0
---hNXNazwk---B--
GET /forum/viewtopic.php?t=5635 HTTP/1.1
Host: example.com
From: googlebot(at)googlebot.com
User-Agent: Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/144.0.7559.132 Mobile Safari/537.36 (compatible; Googlebot/2.1; +http://www.google.com/bot.html)
Accept: text/html,application/xhtml+xml,application/signed-exchange;v=b3,application/xml;q=0.9,*/*;q=0.8
Accept-Encoding: gzip, deflate, br
Amp-Cache-Transform: google;v="1..8"
X-Anubis-Rule: bot/googlebot
X-Anubis-Action: ALLOW
X-Http-Fingerprint-Ja4h: ge11nn080000_3718453403af_000000000000_000000000000
X-Forwarded-For: 127.0.0.1
X-Http-Version: HTTP/1.1
X-Real-Ip: 66.249.68.130

---hNXNazwk---D--

---hNXNazwk---F--
HTTP/1.1 403
Server: nginx/1.28.2
Date: Mon, 02 Mar 2026 01:41:55 GMT
Content-Length: 555
Content-Type: text/html
Connection: keep-alive

---hNXNazwk---H--
ModSecurity: Warning. Matched "Operator `Rx' with parameter `(?i),.*?(?:[\)0-9a-f](?:$|[\"'`](?:$|[^\"'`]+[\"'`])|(?:\r?\n)?\z)|[\"'`][^\"'`]+[\"'`])|[^0-9A-Z_a-z]select.+[^0-9A-Z_a-z]*?from|(?:alter|(?:(?:cre|trunc|upd)at|renam)e|d(?:e(?:lete|sc)|rop)|(?:inser (53 characters omitted)' against variable `REQUEST_HEADERS:User-Agent' (Value: `Mozilla/5.0 (Linux; Android 6.0.1; Nexus 5X Build/MMB29P) AppleWebKit/537.36 (KHTML, like Gecko) Chr (100 characters omitted)' ) [file "/usr/local/etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf"] [line "834"] [id "942200"] [rev ""] [msg "Detects MySQL comment-/space-obfuscated injections and backtick termination"] [data "Matched Data: , like Gecko) Chrome/144.0.7559.132 Mobile Safari/537.36 (compatible; Googlebot/2.1;  http://www.google.com/bot.html) found within REQUEST_HEADERS:User-Agent: Mozilla/5.0 (Linux; Android (173 characters omitted)"] [severity "2"] [ver "OWASP_CRS/4.24.0"] [maturity "0"] [accuracy "0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-sqli"] [tag "paranoia-level/2"] [tag "OWASP_CRS"] [tag "OWASP_CRS/ATTACK-SQLI"] [tag "capec/1000/152/248/66"] [hostname "unix:/var/run/nginx/nginx.sock"] [uri "/forum/viewtopic.php"] [unique_id "177241571537.088552"] [ref "o83,117v68,200t:urlDecodeUni"]
ModSecurity: Access denied with code 403 (phase 2). Matched "Operator `Ge' with parameter `5' against variable `TX:BLOCKING_INBOUND_ANOMALY_SCORE' (Value: `5' ) [file "/usr/local/etc/modsecurity/owasp-modsecurity-crs/rules/REQUEST-949-BLOCKING-EVALUATION.conf"] [line "222"] [id "949110"] [rev ""] [msg "Inbound Anomaly Score Exceeded (Total Score: 5)"] [data ""] [severity "0"] [ver "OWASP_CRS/4.24.0"] [maturity "0"] [accuracy "0"] [tag "anomaly-evaluation"] [tag "OWASP_CRS"] [hostname "unix:/var/run/nginx/nginx.sock"] [uri "/forum/viewtopic.php"] [unique_id "177241571537.088552"] [ref ""]
```

### Your Environment

* CRS version (e.g., v3.3.4): v4.24.0
* Paranoia level setting (e.g. PL1) : PL2
* ModSecurity version (e.g., 2.9.6): 3.0.14
* Web Server and version or cloud provider / CDN (e.g., Apache httpd 2.4.54): nginx
* Operating System and version: FreeBSD 14.3

### Confirmation

[x] I have removed any personal data (email addresses, IP addresses,
    passwords, domain names) from any logs posted.

**Comment by NullIsNot0** [comment]
I also got this FP with Brave browser on macOS when upgraded to CRS 4.24.0. Did not test it on other browsers as reverted back to 4.23.0.
UA: `Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36`

```
[Mon Mar 02 06:51:54.293644 2026] [security2:error] [pid 1228175:tid 1228225] [client CLIENT_IP:CLIENT_PORT] ModSecurity: Warning. Pattern match "(?i),.*?(?:[\\\\)0-9a-f](?:$|[\\"'`](?:$|[^\\"'`]+[\\"'`])|(?:\\\\r?\\\\n)?\\\\z)|[\\"'`][^\\"'`]+[\\"'`])|[^0-9A-Z_a-z]select.+[^0-9A-Z_a-z]*?from|(?:alter|(?:(?:cre|trunc|upd)at|renam)e|d(?:e(?:lete|sc)|rop)|(?:inser|selec)t|load)[\\\\s\\\\x0b]*?\\\\([\\\\s\\\\x0b]*?space[\\ ..." at REQUEST_HEADERS:user-agent. [file "/path/to/coreruleset/v4.x/rules/REQUEST-942-APPLICATION-ATTACK-SQLI.conf"] [line "853"] [id "942200"] [msg "Detects MySQL comment-/space-obfuscated injections and backtick termination"] [data "Matched Data: , like Gecko) Chrome/145.0.0.0 Safari/537.36 found within REQUEST_HEADERS:user-agent: Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/145.0.0.0 Safari/537.36"] [severity "CRITICAL"] [ver "OWASP_CRS/4.24.0"] [tag "application-multi"] [tag "language-multi"] [tag "platform-multi"] [tag "attack-sqli"] [tag "paranoia-level/2"] [tag "OWASP_CRS"] [tag "OWASP_CRS/ATTACK-SQLI"] [tag "capec/1000/152/248/66"] [hostname "DOMAIN"] [uri "/path"] [unique_id "ID_OF_THE_REQUEST"], referer: https://DOMAIN.com
```

**Comment by airween** [comment]
Hi @mhoran,

thanks for reporting this issue (and thanks @NullIsNot0 too)! This feedback is very important to us.

Today there will be the monthly chat, so I'm going to put this on the list and we can discuss.

**Comment by airween** [comment]
@mhoran and @NullIsNot0 - thanks again for this report. We discussed the problem, as you can see @franbuehler assigned the issue to herself, hopefully we can solve this soon, but at late till end of March.

**Comment by franbuehler** [comment]
Commit: https://github.com/coreruleset/coreruleset/commit/6f855a42169ae1f4f94c708d7e36e4ee88930121

Old regex: 
```
(?i),.*?[\"'\)0-9`-f][\"'`](?:[\"'`].*?[\"'`]|(?:\r?\n)?\z|[^\"'`]+)|[^0-9A-Z_a-z]select.+[^0-9A-Z_a-z]*?from|(?:alter|(?:(?:cre|trunc|upd)at|renam)e|d(?:e(?:lete|sc)|rop)|(?:inser|selec)t|load)[\s\x0b]*?\([\s\x0b]*?space[\s\x0b]*?\(
```

New regex: 
```
(?i),.*?(?:[\)0-9a-f](?:$|[\"'`](?:$|[^\"'`]+[\"'`])|(?:\r?\n)?\z)|[\"'`][^\"'`]+[\"'`])|[^0-9A-Z_a-z]select.+[^0-9A-Z_a-z]*?from|(?:alter|(?:(?:cre|trunc|upd)at|renam)e|d(?:e(?:lete|sc)|rop)|(?:inser|selec)t|load)[\s\x0b]*?\([\s\x0b]*?space[\s\x0b]*?\(
```

Minimal strings that match the new regex:
* `, like Gecko) Chrome`
* `, se`
* `, f`
* `, f'`
* `, feeeeeeee`
* `, like Gecko) Chrome/142.0.0.0 Safari/537.36x22`

The problem is this part of the regex:
`(?i),.*?(?:[\)0-9a-f]$)`

I don't have a solution yet; I need to think about it. I just want to note some of my findings.

**Comment by franbuehler** [comment]
Suggestion to extend the original, old regex:

Instead of (first part only):
```
(?i),.*?[\"'\)0-9`-f][\"'`](?:[\"'`].*?[\"'`]|(?:\r?\n)?\z|[^\"'`]+)
```

We extend it with 
```
[\"'`]
```
at the end:

```
(?i),.*?[\"'\)0-9`-f][\"'`](?:[\"'`].*?[\"'`]|(?:\r?\n)?\z|[^\"'`]+[\"'`])
```

so that a second `'` has to follow. This would guarantee that `999, rue d'Arlon` doesn't match.

The tests still match because they contain `SELECT * FROM` which is in the second part of the regex.

Full proposed regex with tests:
https://regex101.com/r/JqtnKy/1

**Comment by EsadCetiner** [comment]
@franbuehler Are you able to open a PR to fix this issue? Right now PL-2 and higher is unusable since pretty much all User-Agents are blocked.

This might be resolved by https://github.com/coreruleset/coreruleset/pull/4525

**Comment by theseion** [comment]
I've opened a PR with a fix and new test cases.
